src/vector_windows.rs

```diff
@@ -0,0 +1,247 @@
+use crate::service::start_service;
+use std::{ffi::OsString, sync::mpsc, time::Duration};
+use windows_service::service::{
+    ServiceControl, ServiceExitCode, ServiceState, ServiceStatus, ServiceType,
+};
+use windows_service::{
+    define_windows_service, service::ServiceControlAccept,
+    service_control_handler::ServiceControlHandlerResult, service_dispatcher, Result,
+};
+
+const SERVICE_NAME: &str = "vector";
+const SERVICE_TYPE: ServiceType = ServiceType::OWN_PROCESS;
+
+pub mod service_control {
+    use windows_service::service::{ServiceErrorControl, ServiceInfo, ServiceStartType};
+    use windows_service::{
+        service::{ServiceAccess, ServiceState},
+        service_manager::{ServiceManager, ServiceManagerAccess},
+        Result,
+    };
+
+    use crate::vector_windows::{SERVICE_NAME, SERVICE_TYPE};
+    use std::ffi::OsString;
+    use std::time::Duration;
+    use std::{fmt, thread};
+
+    #[derive(Debug, Clone, PartialEq)]
+    pub enum ControlAction {
+        Install,
+        Uninstall,
+        Start,
+        Stop,
+    }
+
+    impl fmt::Display for ControlAction {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            write!(f, "{:?}", self)
+        }
+    }
+
+    struct ServiceDefinition {
+        pub name: OsString,
+        pub display_name: OsString,
+        pub description: OsString,
+
+        pub executable_path: std::path::PathBuf,
+        pub launch_arguments: Vec<OsString>,
+    }
+
+    impl std::str::FromStr for ControlAction {
+        type Err = String;
+
+        fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+            match s {
+                "install" => Ok(ControlAction::Install),
+                "uninstall" => Ok(ControlAction::Uninstall),
+                "start" => Ok(ControlAction::Start),
+                "stop" => Ok(ControlAction::Stop),
+                _ => Err(format!("invalid option {} for ControlAction", s)),
+            }
+        }
+    }
+
+    pub fn configure_service(
+        action: ControlAction,
+        config_paths: Vec<std::path::PathBuf>,
+    ) -> Result<()> {
+        info!("Configuring vector as a service. action={}", action);
+
+        let current_exe = ::std::env::current_exe().unwrap();
+
+        let arguments = config_paths
+            .iter()
+            .flat_map(|p| vec![OsString::from("--config"), p.as_os_str().into()])
+            .collect::<Vec<OsString>>();
+
+        let service = ServiceDefinition {
+            name: OsString::from(SERVICE_NAME),
+            display_name: OsString::from("Vector Service"),
+
+            description: OsString::from("Vector is a lightweight, ultra-fast, open-source tool for building observability pipelines."),
```

Comment:
We use https://docs.rs/built/0.4.2/built/ which can provide these. `PKG_DESCRIPTION` in this case